LanguageFeatures/Patterns/execution_pattern_for_statement_A01_t01.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// A statement of the form:
+///
+/// for (<patternVariableDeclaration>; <condition>; <increment>) <statement>
+///
+/// Is executed like a traditional for loop though is more likely to declare
+/// multiple variables.
+///
+/// @description Check that pattern-for statement is executed like a traditional
+/// for loop. Test variable and identifier subpatterns of parenthesized pattern
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/expect.dart";
+
+main() {
+  String log = "";
+  for (var (int i) = 0; i < 3; (i) += (i + 1)) {
```

Comment:
That `+=` token would be a syntax error. ;-)

But `(i) = i + 1` would suffice, anyway.